core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -1041,10 +1060,20 @@ void cleanupExpiredRemoteLogSegments() throws RemoteStorageException, ExecutionE
             // and delete them accordingly.
             // If the follower HAS NOT picked up the changes, and they become the leader then they will go through this process
             // again and delete them with the original deletion reason i.e. size, time or log start offset breach.
+            BrokerTopicMetrics brokerTopicMetrics = brokerTopicStats.topicStats(topicIdPartition.topic());
+            int partition = topicIdPartition.partition();
+            brokerTopicMetrics.recordRemoteDeleteBytesLag(partition, sizeOfDeletableSegmentsBytes);
+            int segmentsLeftToDelete = segmentsToDelete.size();
+            brokerTopicMetrics.recordRemoteDeleteSegmentsLag(partition, segmentsLeftToDelete);
```

Comment:
Should we also add the segments in L1095?

Reply:
You are absolutely correct - added in the newest version